### PR TITLE
fix(cluster.py): do not print false errors not finding 'cpu_set' value

### PR DESCRIPTION
**Pull request by vponomaryov**
Read `/etc/scylla.d/cpuset.conf` file contents and parse it in python
instead of using 'grep' command which returns exit code '1' when no match is found.
It will allow us to avoid following false error messages:

    > Command: 'grep "^CPUSET" /etc/scylla.d/cpuset.conf'
    >
    > Exit code: 1
    >
    > Stdout:
    >
    >
    >
    > Stderr:
    >
    >
    >

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -556,7 +556,11 @@ def cpuset(self):
             # or
             #   'CPUSET="--cpuset 1-7,9-15,17-23,25-31 "'
             # And so on...
-            grep_result = self.remoter.run('grep "^CPUSET" /etc/scylla.d/cpuset.conf')
+            # NOTE: consider as error only 2+ exit codes because the '1' exit code in case of 'grep'
+            #       means 'match not found' which happens always on K8S.
+            grep_result = self.remoter.run(
+                "grep \"^CPUSET\" /etc/scylla.d/cpuset.conf"
+                " || EXIT_CODE=$? ; if [[ $EXIT_CODE != 1 ]]; then exit $EXIT_CODE; fi")
```
Comment (fgelcer) [inline sdcm/cluster.py:563]: shound't it be `[[ $EXIT_CODE -eq  1 ]]` ?

Comment (fruch) [inline sdcm/cluster.py:563]: wouldn't it be clearer to do all this logic on the python level ?

**Review by fruch — CHANGES_REQUESTED** [review]
see if you can move the logic from bash to python

**Review by fruch — APPROVED** [review]
LGTM

**Review by soyacz — APPROVED** [review]

**Pull request merged**